spec/features/smart_answer/business_coronavirus_support_finder_flow_spec.rb

```diff
@@ -0,0 +1,102 @@
+      self_assessment_july_2020: "Are you due to pay a Self Assessment payment on account by 31 July 2020?",
+      self_employed: "Are you self-employed?",
+      what_size_was_your_buisness: "What size was your business as of 28 February?",
+      where_is_your_business_based: "Where is your business based?",
+      results: "Support you may be entitled to",
+    }
+  end
+
+  before do
+    stub_content_store_has_item("/business-coronavirus-support-finder")
+  end
+
+  # Only the non_domestic_property question causes a flow path change.
+  # For all other questions, any of the available answers would result in the same next page.
+  scenario "Answers for business that does not have non domestic property" do
+    visit "/business-coronavirus-support-finder"
+    expect(page).to have_h1(headings[:flow_title])
+
+    click_govuk_start_button
+    expect(page).to have_h1(headings[:business_based])
+
+    choose "England"
+    click_button "Next step"
+    expect(page).to have_h1(headings[:business_size])
+
+    choose "0 to 249 employees"
+    click_button "Next step"
+    expect(page).to have_h1(headings[:annual_turnover])
+
+    choose "My business is a start-up and is pre-revenue"
+    click_button "Next step"
+    expect(page).to have_h1(headings[:paye_scheme])
+
+    choose "Yes"
+    click_button "Next step"
+    expect(page).to have_h1(headings[:self_employed])
+
+    choose "Yes"
+    click_button "Next step"
+    expect(page).to have_h1(headings[:non_domestic_property])
+
+    choose "My business does not have any non-domestic property"
+    click_button "Next step"
+    expect(page).to have_h1(headings[:flow_title])
+    expect(page).to have_selector("h2", text: headings[:results])
+  end
+
+  scenario "Answers for business that has non domestic property" do
+    visit "/business-coronavirus-support-finder"
+    expect(page).to have_h1(headings[:flow_title])
+
+    click_govuk_start_button
+    expect(page).to have_h1(headings[:business_based])
+
+    choose "England"
+    click_button "Next step"
+    expect(page).to have_h1(headings[:business_size])
+
+    choose "0 to 249 employees"
+    click_button "Next step"
+    expect(page).to have_h1(headings[:annual_turnover])
+
+    choose "My business is a start-up and is pre-revenue"
+    click_button "Next step"
+    expect(page).to have_h1(headings[:paye_scheme])
+
+    choose "Yes"
+    click_button "Next step"
+    expect(page).to have_h1(headings[:self_employed])
+
+    choose "Yes"
+    click_button "Next step"
+    expect(page).to have_h1(headings[:non_domestic_property])
+
+    choose "Under £51,000"
+    click_button "Next step"
+    expect(page).to have_h1(headings[:sectors])
+
+    check "Nurseries"
+    click_button "Next step"
+    expect(page).to have_h1(headings[:rate_relief_march_2020])
+
+    choose "Yes"
+    click_button "Next step"
+    expect(page).to have_h1(headings[:flow_title])
+    expect(page).to have_selector("h2", text: headings[:results])
```

Comment:
Nit: Hmm this feels like its going to be repeated pattern for most flows... I wonder if a helper method might be useful for dealing with the "Check heading, choose option, click next" pattern?

Reply:
I really think that's a bad idea.

After having worked on other GDS projects where every call is abstracted out to a method I purposely left this reasonably untouched raw capybara code. Yes it is a little verbose, but it's also very clear exactly what's going on and what's being tested. If you start abstracting methods out too far, trying to work out what a spec is doing is a nightmare because you have to search through loads of support files to work out that things like `click_next_step` is just doing `click_button "Next step"`.

I made two small concessions - `click_govuk_start_button` (because the raw capybara isn't clear), and `have_h1` which has a very obvious, limited, and clear behaviour.

Oh - and note that line 93 uses `check` rather than `choose`, so one method wouldn't work for all of the steps.

Reply:
`have_h1` has been removed.